eng/scripts/Language-Settings.ps1

```diff
@@ -172,58 +172,97 @@ function Get-javascript-GithubIoDocIndex()
-# For "latest", we simply set a target package name
-# For "preview", we add @next to the target package name
-function Update-javascript-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId = $null)
-{
-  $pkgJsonLoc = (Join-Path -Path $ciRepo -ChildPath $locationInDocRepo)
-
-  if (-not (Test-Path $pkgJsonLoc))
-  {
-    Write-Error "Unable to locate package json at location $pkgJsonLoc, exiting."
-    exit(1)
+# "@azure/package-name@1.2.3" -> "@azure/package-name"
+function Get-PackageNameFromDocsMsConfig($DocsConfigName) { 
+  if ($DocsConfigName -match '^(@?.+?)(@(.*))?$') { 
+    return $Matches[1]
   }
 
-  $allJson = Get-Content $pkgJsonLoc | ConvertFrom-Json
+  throw "Cannot find package $DocsConfigName"
+}
 
-  $visibleInCI = @{}
+function Update-javascript-DocsMsPackages($DocsRepoLocation)
+{
+  $publishedPackages = Get-CSVMetadata `
+  | Where-Object { $_.New -eq 'true' -and $_.Hide -ne 'true' }
+
+  # Update "Latest" packages
+  $latestConfigPath = Join-Path $DocRepoLocation '/ci-configs/packages-latest.json' -Resolve
+  Write-Host "Updating latest configuration: $latestConfigPath"
+  $latestPackages = Get-Content $latestConfigPath -Raw | ConvertFrom-Json
+
+  $gaPackages = $publishedPackages | Where-Object { $_.VersionGA.trim() }
+  foreach ($package in $gaPackages) { 
+    if ($latestPackages.npm_package_sources.name -contains $package.Package) { 
+      # Onboarded packages already contains this package, skip
+      Write-Host "Keeping onboarded package $($package.Package)"
+      continue
+    }
 
-  for ($i = 0; $i -lt $allJson.npm_package_sources.Length; $i++)
-  {
-    $pkgDef = $allJson.npm_package_sources[$i]
-    $accessor = ($pkgDef.name).Replace("`@next", "")
-    $visibleInCI[$accessor] = $i
+    Write-Host "Add GA package: $($package.Package)"
+    $latestPackages.npm_package_sources += @{ name = $package.Package }
   }
 
-  foreach ($releasingPkg in $pkgs)
-  {
-    $name = $releasingPkg.PackageId
-
-    if ($releasingPkg.IsPrerelease)
-    {
-      $name += "`@next"
+  $latestPackages | ConvertTo-Json -Depth 100 | Set-Content $latestConfigPath
+
+  # Update "preview" packages
+  $previewConfigPath = Join-Path $DocRepoLocation '/ci-configs/packages-preview.json' -Resolve
+  Write-Host "Updating preview configuration: $previewConfigPath"
+  $previewPackages = Get-Content $previewConfigPath -Raw | ConvertFrom-Json
+
+  $outputPackages = @()
+  foreach ($package in $previewPackages.npm_package_sources) {
+    # Do not filter by GA/Preview status because we want differentiate between
+    # tracked and non-tracked packages
+    $matchingPublishedPackages = $publishedPackages `
+    | Where-Object { $_.Package -eq (Get-PackageNameFromDocsMsConfig $package.name) }
+
+    # If this package does not match any published packages keep it in the list.
+    # This handles packages which are not tracked in metadata but still need to
+    # be built in Docs CI.
+    if (-not $matchingPublishedPackages) {
```

Comment:
If you change the above where-object to where then you will just check the count here. 
```suggestion
    if ($matchingPublishedPackages.Count -eq 0) {
```